_posts/2025-01-31-wildfly-domain-usage.adoc

```diff
@@ -0,0 +1,497 @@
+[Host Controller] 17:06:51,139 INFO  [org.jboss.remoting] (MSC service thread 1-7) JBoss Remoting version 5.0.30.Final
+[Host Controller] 17:06:52,510 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0062: Http management interface listening on http://192.168.0.115:9990/management and https://192.168.0.115:-1/management
+[Host Controller] 17:06:52,510 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0053: Admin console listening on http://192.168.0.115:9990 and https://192.168.0.115:-1
+[Host Controller] 17:06:52,546 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly 35.0.0.Final (WildFly Core 27.0.0.Final) (Host Controller) started in 3066ms - Started 70 of 71 services (14 services are lazy, passive or on-demand) - Host Controller configuration files in use: domain.xml, host.xml - Minimum feature stability level: community
+----
+
+From the above server log output, I can see the server-bound address is `192.168.0.115` instead of the default `127.0.0.1`. Then I opened another terminal window and ran the following command in the `bin` directory of the WildFly server:
+
+[source,bash]
+----
+$ ./add-user.sh -u admin -p 123
+Added user 'admin' to file '/wildfly-35.0.0.Final/standalone/configuration/mgmt-users.properties'
+Added user 'admin' to file '/wildfly-35.0.0.Final/domain/configuration/mgmt-users.properties'
+----
+
+From the above command output, we can see the `admin` user is added and its password is `123`. This added user will be used on the Secondary Host Controller to connect to the Primary Host controller.
+
+Please note that the above command didn't generate a secret value related to the generated `admin` user as described you may find in the other online materials, which is already deprecated since WildFly 25. So the following `secret` configuration on the Secondary Host Controller is already deprecated:
+
+[source,xml]
+----
+<server-identities>
+  <secret value="..." />
+</server-identities>
+----
+
+As the Primary Host Controller is started on Machine A, now we can check the WildFly server configuration on Machine B. As planned, the Host Controller is named `secondary`, and it will be connected to the Primary Host Controller on Machine A, which acts as the domain controller.
+
+Now we can check the configuration on the Secondary Host Controller. The Secondary Host Controller on Machine B will use the `host-secondary.xml` file as its configuration file. Please note that to edit the `host-secondary.xml` file in CLI, you should use this command to connect to the embedded server and use the correct host controller file:
+
+[source,bash]
+----
+[disconnected /] embed-host-controller --host-config=host-secondary.xml
+[domain@embedded /]
+----
+
+We need to make some modifications to this file. First I need to add the `name` property to the `host` section:
+
+[source,xml]
+----
+<host xmlns="urn:jboss:domain:community:20.0" name="secondary">
```

Comment:
@jamezp Is there a way to configure the host name here via CLI command?

Reply:
@yersan @jamezp I tried to use the following command to edit the host name on the secondary host but failed:

```bash
[domain@embedded /] /host=current-host:write-attribute(name=name, value=secondary)
Failed to get the list of the operation properties: "WFLYDC0032: Operation read-operation-description for address [("host" => "current-host")] can only be handled by the Domain Controller; this host is not the Doma
in Controller"
```

I guess this means I can't directly edit the host name on the secondary host by using the embed server command am I correct?

Reply:
The only difference I could see is that in your command it's: `/host=localhost`. I'll try with this value, thanks for checking!

Reply:
After doing some experiments, I found the current host name affects the output and that's the root cause:

<img width="1904" alt="image" src="https://github.com/user-attachments/assets/87dc1b9f-cc7c-4371-9743-4c4f37f152fc" />

Now I understand how it works. Thanks for the help Yeray!